app/Main.hs

```diff
@@ -188,6 +190,11 @@ main = do
       pathNotFound :: Member (Embed IO) r => QualifiedPath Path -> Sem r ()
       pathNotFound path = printError $ "Entry or directory not found at '" +| path |+ "'."
 
+      samePaths :: Member (Embed IO) r => QualifiedPath Path -> Path -> Sem r ()
+      samePaths old@(QualifiedPath backend _) newPath =
+        printError $ "'" +| old |+ "' and '" +|
+          (QualifiedPath backend newPath) |+ "' are the same path."
```

Comment:
Instead of extracting the backend from `oldQPath :: QualifiedPath Path` and adding it to `newPath :: Path`, we could just save the qualified `newQPath :: QualifiedPath Path` inside `CPRSamePath`?

Or even just save one of the two paths (since they're the same anyway)?

Reply:
Oh, really, I will just leave one of them. I think in the beginning I decided to pass two, because I thought they might have different string representations 🤣